jwst/pipeline/calwebb_detector1.py

```diff
@@ -73,6 +73,24 @@ def process(self, input):
 
         log.info('Starting calwebb_detector1 ...')
 
+        if self.ramp_fit.algorithm.upper() == "LIKELY":
+            with datamodels.RampModel(input) as ramp:
+                ngroups = ramp.data.shape[1]
+            if ngroups < 4:
+                log_msg = "For the LIKELY ramp fit algorithm NGROUPS must be"
+                log_msg = f"{log_msg} at least 4.  It NGROUPS = {ngroups}."
+                log_msg = f"{log_msg}  Defaulting to OLS_C.  Also, the normal"
+                log_msg = f"{log_msg} pipeline jump detection algorithm will be used."
```

Comment:
Oh, sorry, I misunderstood -- I see what you're doing now, with inserting the previous fragment at the beginning of the message.

Still, this would be more efficiently composed as a single string e.g.
```
                log.warning(f"For the LIKELY ramp fit algorithm NGROUPS must be "
                            f"at least 4.  Input has NGROUPS = {ngroups}.")
                log.warning("Defaulting to OLS_C.  Also, the normal "
                            "pipeline jump detection algorithm will be used.")
```